From b2422dccdb1beba56da778cd4881ba72f40dc450 Mon Sep 17 00:00:00 2001
From: Michael Catanzaro <mcatanzaro@redhat.com>
Date: Thu, 6 Nov 2025 03:44:29 -0600
Subject: [PATCH] gfileutils: fix computation of temporary file name

We need to ensure that the value we use to index into the letters array
is always positive.

Fixes #3716

Signed-off-by: Vamsi Krishna Brahmajosyula <vamsi-krishna.brahmajosyula@broadcom.com>
---
 glib/gfileutils.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/glib/gfileutils.c b/glib/gfileutils.c
index 44c22f3..fa03258 100644
--- a/glib/gfileutils.c
+++ b/glib/gfileutils.c
@@ -1477,9 +1477,9 @@ get_tmp_file (gchar            *tmpl,
   static const char letters[] =
     "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
   static const int NLETTERS = sizeof (letters) - 1;
-  glong value;
-  gint64 now_us;
-  static int counter = 0;
+  guint64 value;
+  guint64 now_us;
+  static guint counter = 0;
 
   g_return_val_if_fail (tmpl != NULL, -1);
 
@@ -1498,7 +1498,7 @@ get_tmp_file (gchar            *tmpl,
 
   for (count = 0; count < 100; value += 7777, ++count)
     {
-      glong v = value;
+      guint64 v = value;
 
       /* Fill in the random bits.  */
       XXXXXX[0] = letters[v % NLETTERS];
-- 
2.51.1

